service/views.py

```diff
@@ -4,21 +4,43 @@
 from django.contrib.auth import authenticate, login, logout
 from django.contrib.auth.decorators import login_required
 from django.db.models import Avg, F
+from rest_framework.decorators import api_view
 # Create your views here.
 
 
 @login_required(login_url='login')
+@api_view(['GET'])
 def home(request):
-    systems = request.user.system_set.all()
-    technician = request.user
+    """Show systems assigned to technician"""
     technician = request.user
+    if request.query_params.get('q') is not None:
+        q = request.query_params.get('q')
+        systems = request.user.system_set.all().filter(name__icontains=q)
```

Comment:
Turns out search could be implemented in better way https://www.django-rest-framework.org/api-guide/filtering/#searchfilter